spopt/region/maxp.py

```diff
@@ -10,27 +9,145 @@
 __email__ = "sjsrey@gmail.com"
 
 from ..BaseClass import BaseSpOptHeuristicSolver
-from .base import (
-    w_to_g,
-    move_ok,
-    ok_moves,
-    region_neighbors,
-    _centroid,
-    _closest,
-    _seeds,
-    is_neighbor,
-)
 
 from scipy.spatial.distance import pdist, squareform
-import geopandas as gp
+from scipy.spatial import KDTree
+from scipy.sparse.csgraph import connected_components
+import libpysal
 import numpy as np
 from copy import deepcopy
-from scipy.sparse.csgraph import connected_components
 
 ITERCONSTRUCT = 999
 ITERSA = 10
 
 
+def infeasible_components(gdf, w, threshold_var, threshold):
+    """Identify infeasible components.
+
+    Parameters
+    ----------
+    gdf : geopandas.GeoDataFrame, required
+        Geodataframe containing original data
+
+    w : libpysal.weights.W, required
+        Weights object created from given data
+
+    attrs_name : list, required
+        Strings for attribute names to measure similarity
+        (cols of ``geopandas.GeoDataFrame``).
+
+    threshold_var : string, requied
+        The name of the spatial extensive attribute variable.
+
+    threshold : {int, float}, required
+        The threshold value.
+
+    Returns
+    -------
+    list of infeasible components
+    """
+    components = np.unique(w.component_labels)
+    gdf['_components'] = w.component_labels
+    gb = gdf.groupby(by='_components').sum()
+    gdf.drop(columns="_components", inplace=True)
+    if gb[threshold_var].min() < threshold:
+        l = gb[gb[threshold_var]< threshold]
+        return l.index.values.tolist()
+    return []
+
+
+def plot_components(gdf, w):

```

Comment:
nice, this can be a really useful check. It might be nice to lift it up into `lib`...?

Reply:
I was thinking the same. 

What we could do is leave this here for now, then as part of the next release of lib, migrate this over and have spopt import from lib.